src/test/java/com/google/sps/BookListsTest.java

```diff
@@ -0,0 +1,145 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.*;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+import com.google.sps.servlets.BookListsServlet;
+import com.google.sps.data.BookList;
+import com.google.gson.Gson;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+@RunWith(JUnit4.class)
+public final class BookListsTest {
+    private final LocalServiceTestHelper helper =  new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()); 
+
+    private BookListsServlet servlet;
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+
+        servlet = new BookListsServlet();
+        servlet.init();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void testDoGetWithNoLists() throws Exception {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter("topic")).thenReturn("random");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+        when(request.getContentType()).thenReturn("application/json");
+
+        servlet.doGet(request, response);
+
+        verify(request, times(1)).getParameter("topic");
+
+        String expected = "[]";
+      
+        writer.flush();
+
+        // No lists should have been returned
+        Assert.assertTrue(stringWriter.toString().contains(expected));
+    }
+
+    @Test
+    public void testDoGetWithLists() throws Exception {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        ArrayList<String> books = new ArrayList<String>(Arrays.asList("Book 1 by Random Person", "Book 2 by Anonymous"));
+
+        // Adds goal entity to the local datastore
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity listEntity = new Entity("BookList");
+        listEntity.setProperty("tutorID", "123");
+        listEntity.setProperty("books", books);
+        listEntity.setProperty("name", "My Math Books");
+        listEntity.setProperty("topic", "math");
+
+        datastore.put(listEntity);
```

Comment:
Not super necessary but this could be a sample data function that adds book lists ti the datastore (like addStudentsToDatastore and addTutorsToDatastore)

Reply:
I think I'm only adding one book at a time, so I don't think it's super necessary right now. If we update the tests to add more than one at a time, making a separate method would be a good idea.